app/eme/window.js

```diff
@@ -57,6 +65,7 @@ class Window {
     win.$state = {
       unsaved: 0
     }
+    win.configs = config
```

Comment:
I guess all these shared data should be named `$xxx`, eg: `win.$config = config`